### DISCUSSION TITLE
Incomplete video download from YouTube

### DISCUSSION THREAD

**Discussion by txr13** [aclap-dev/video-downloadhelper#950] [Help and Q&A]
### Your question

I am attempting to download an unlisted video from YouTube. The total duration of the video is 9:51:44, and it is fully accessible when played via YouTube itself.

When downloaded by VDH, the display shows a duration of exactly 4:00:00. The downloaded version is the _last_ four hours of the stream--all earlier content is missing, even though the page starts the video at the very beginning.

I would appreciate any suggestions that could be offered, and apologize if I'm just missing something simple.

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"58be6eafd9904d11"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "valid_license_message_has_been_displayed": true,
    "download_directory": "C:\\Users\\User\\dwhelper",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\User\\dwhelper"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 24,
    "successfull_dl": 3
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

One simple thing you are missing is the URL of the video.  I'm not sure what you mean by "unlisted."  Is it a private video, available only to certain users?  Things that can't be accessed by just any old user are pretty much impossible to investigate.  Even though this thing is nearly 10 hours long, YouTube typically gives decent enough download speed that it wouldn't take an unreasonable time to download it.  Plus, their content is highly compressed so the file would be smaller than something comparable on a more standards-compliant web site.  I would try to download it if I knew what it was.  But if it's private, I won't be able to help you.  I will say that I have recorded live golf tournaments from TV broadcasts that were about 8 hours long . . . back in the days before NBC decided to put all their content behind DRM protection, so it's been a while since I was able to get something that big.  But I do download operas on occasion (not from YouTube) & those can sometimes run to 4 or 5 hours.  So the fact that you aren't getting the whole video is rather surprising.

**Reply by txr13** [reply]
Sorry! I've updated my original post with the URL in question. It is not a private video, merely unlisted. (Meaning it's not available through searching for it, only by those who have the direct URL. But the URL itself should be accessible by anybody, as it is not DRM'd or protected in any other way I know of.)

**Reply by GrampaWildWilly** [reply]
Oh good.  Let's see what I can figure out . . .

**Reply by txr13** [reply]
And you know, I just went back and rechecked the URL. VDH now shows the entire duration available, even though I tried it three separate times earlier, in both regular and incognito windows. Let me try downloading it again...

**Reply by GrampaWildWilly** [reply]
I've got it running now.  I'm getting about 14 million bytes per second download speed.  VDH is telling me I've gotten one quarter of it in about 3 minutes.  So this is looking like I'll get it in about 15 minutes, max.  That is, assuming YouTube doesn't get a wild hair up its butt & decide to throttle me back to some slower speed.  So let's see what's what in a few minutes.

**Reply by txr13** [reply]
VDH reported that it was unable to download the file and asked me to report it... and yet the full file did complete downloading, and I've checked it and verified everything is intact and plays as it should.

Something's odd about that, but I'll put it down to large sizes and the vagaries of YouTube. Thank you for being willing to help!

**Reply by GrampaWildWilly** [reply]
OK.  Here's what I've gotten:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/caf49df0-b912-41d2-99fe-f246fd3dc7db)

Today's download speed is pretty good for YouTube.  Barely 12 minutes for a nearly 10-hour show is not too shabby at all.  As expected, the video bit rates are pretty crappy.  Decent 1920x1080 content should have bit rates on the order of 5000kbps.  On the other hand, since this item has less than half that, the file is commensurately probably about half the size it would be otherwise.  The audio bit rate being 127kbps is just sort of average.  This is a common bit rate you'll see in lots of content around the web.  The sampling rate of 44.1kHz is also quite common.

I played the thing, although at 10 hours long, you can be sure I didn't (& will never) sit & watch this thing all the way through.  One thing that kind of bloats up its duration is all the wasted time.  It's about 37 minutes in before the show even starts.  Then there's these long breaks scattered throughout.  I'm guessing they could probably have cut this down to 7, maybe 6 hours.  But whatever.  It appears I've gotten the full show.

Now here's an observation I've made in the past.  I like to download (from YouTube) highlights clips of certain hockey games of a certain team whose name you can guess from my avatar.  Not at this time of year, of course.  Of course, the clips don't become available until after a certain delay after the end of the game.  Occasionally, I happen to find the clips just a minute or two after they are posted.  At that point, the only resolution being offered is something offensively low like 360p or 240p.  But if I wait a few minutes, oops, here come all the other resolutions, in particular the 1280x720 that seems to be the highest the NHL offers.

So if you jump the gun, you don't get what will ultimately be the final product.  I am going to put forward the hypothesis that you saw something like this with this archived livestream.  You just found it before it had been 100% posted on YouTube.  It's certainly all there now.

So I would take this as an object lesson & just wait a bit until the whole thing has been posted.  Next time you see this sort of anomaly, just check back 5, 10, 15 minutes later.  Keep checking until the whole thing is finally available.  If I'm doing the math right, and interpreting the information on the page right, the livestream began about 8:00 AM earlier today my time.  If it ran for 10 hours, it completed about 6:00 PM my time.  That's about 3 hours ago.  I would not be surprised if it took as long as an hour for this thing to get fully posted.  So if you looked for this before about 7:00 PM my time & saw it wasn't all there, you were just too early.  You should have just waited a while.  Now, "while" isn't exactly a precise mathematical time to wait.  But I don't think we have all the information to make a more accurate guess than "while."  For any future, similar case, this is what I would recommend.

**Reply by GrampaWildWilly** [reply]
An error.  Hmmmm........ I didn't get an error.  Are you sure it was from your present download of this video & not some earlier error on something else that failed?  The errors stick around in the VDH menu until you clear them via any of the various available methods.  So you need to be sure it was a new error & not a leftover from some time ago.

I have to say I have mixed feelings about this new error reporting.  On the one hand, doing it this way probably reduces clutter in this forum somewhat.  Instead of people posting their error reports here, they go to a data base that @paulrouget Paul looks at.  The traffic here indeed has fallen a bit since this new approach went into effect.

On the other hand, it makes it utterly impossible for any of us civilians to attempt to diagnose our own problems.  There are other discussions in here where people have complained that they can't connect the errors to the particular downloads that might have failed.  If you are doing a bunch of downloads, it does get rather messy keeping track of what succeeded & what failed.  This new, rather cryptic way of error reporting doesn't help much.

So if your download succeeded, I would ignore the error you got rather than posting it.

And oh by the way.  I probably should have mentioned this sooner.  My operating environment is Windows 11 Pro 64-bit version 23H2 build 22631.3810, Firefox 127.0.2 64-bit, licensed VDH 9.0.2.12a1 beta, CoApp 2.0.19.

**Reply by txr13** [reply]
Oh, I am quite certain it was from the present download and no previous attempts. I was checking and clearing the "completed" downloads after every attempt, to make sure I didn't have any detritus hanging around. Even so, since I got the file fully downloaded anyway, I am inclined to simply ignore the issue (and in fact, I already cleared it without reporting it).

Thank you again for your assistance!
